ext/opcache/jit/zend_jit_x86.dasc

```diff
@@ -15015,6 +15024,43 @@ static int zend_jit_fetch_constant(dasm_State          **Dst,
 	return 1;
 }
 
+static int zend_jit_count(dasm_State **Dst, const zend_op *opline, const zend_op_array *op_array, uint32_t op1_info)
+{
+	zend_jit_addr res_addr = RES_ADDR();
+
+	if (opline->op1_type == IS_CONST) {
+		zval *zv;
+		size_t count;
+
+		zv = RT_CONSTANT(opline, opline->op1);
+		ZEND_ASSERT(Z_TYPE_P(zv) == IS_ARRAY);
+		count = zend_array_count(Z_ARRVAL_P(zv));
+
+		|	SET_ZVAL_LVAL res_addr, count
+		|	SET_ZVAL_TYPE_INFO res_addr, IS_LONG
+	} else {
+		zend_jit_addr op1_addr = OP1_ADDR();
+
+		ZEND_ASSERT((op1_info & (MAY_BE_UNDEF|MAY_BE_ANY|MAY_BE_REF)) == MAY_BE_ARRAY);
+		// Note: See the implementation of ZEND_COUNT in Zend/zend_vm_def.h - arrays do not contain IS_UNDEF starting in php 8.1+.
+
+		|	GET_ZVAL_PTR FCARG1a, op1_addr
+		|	mov r0, [FCARG1a + offsetof(HashTable, nNumOfElements)]
+		|	SET_ZVAL_LVAL res_addr, r0
+		|	SET_ZVAL_TYPE_INFO res_addr, IS_LONG
+		if (opline->op1_type & (IS_VAR|IS_TMP_VAR)) {
+			// Temporary expressions need to be freed.
+			// Anything containing objects or resources might throw when freed due to the destructors.
+			zend_bool gc = (op1_info & (MAY_BE_ARRAY_OF_OBJECT|MAY_BE_ARRAY_OF_ARRAY|MAY_BE_ARRAY_OF_RESOURCE|MAY_BE_ARRAY_OF_REF)) != 0;
+			|	ZVAL_PTR_DTOR op1_addr, op1_info, gc, 0, opline
+			if (gc) {
```

Comment:
I believe the common practice here is to determine this using zend_may_throw (passed in).

Reply:
I need to re-add this, I was running into issues because nNumOfElements was 32-bit when working on the tracing jit

Reply:
Done, could use another look